blockchain/pool.go

```diff
@@ -170,7 +170,7 @@ func (pool *BlockPool) IsCaughtUp() bool {
 
 	// some conditions to determine if we're caught up
 	receivedBlockOrTimedOut := (pool.height > 0 || time.Since(pool.startTime) > 5*time.Second)
-	ourChainIsLongestAmongPeers := pool.maxPeerHeight == 0 || pool.height >= pool.maxPeerHeight
+	ourChainIsLongestAmongPeers := pool.maxPeerHeight == 0 || pool.height >= (pool.maxPeerHeight-1)
```

Comment:
Because we need two blocks to sync one. ie to sync to height H we need block H+1

Reply:
```suggestion
	// we subtract -1 from pool.maxPeerHeight because we need two blocks to sync one. ie to sync to height H we need block H+1
	ourChainIsLongestAmongPeers := pool.maxPeerHeight == 0 || pool.height >= (pool.maxPeerHeight-1)
```

Reply:
Updated the comment